src/tutorial/index.rst

```diff
@@ -435,30 +421,40 @@ Below you can see how the message output function may be assembled. Normally, ag
 
-The only way to access spatial messaging types is via an iterator, which returns all messages in a Moore neighbourhood about the provided search location. This means, that all messages within the originally specified search radius will be returned, however it is necessary for the user to filter out the small number of messages which fall outside of this radius that will also be returned. Furthermore, agent's will also be returned their own message, so may wish to filter out that too.
+The only way to access spatial messaging types is via an iterator, which returns all messages in a Moore neighbourhood (discretised by the message radius) about the provided search location. This means, that all messages within the originally specified search radius will be returned, however it is necessary for the user to filter out messages which are contained within the Moor neighbour but fall outside of this radius that are also returned. Furthermore, agent's will also be returned their own message, so may wish to filter by outputting the originating agent id to the message.
 
-The spatial message iterator is accessed using :func:`FLAMEGPU->message_in()<flamegpu::MessageSpatial2D::In::operator()>`, this takes two ``float`` parameters specifying the search origin. Normally this will be passed directly to a C++ range-based for loop (e.g. ``or (const auto &message : FLAMEGPU->message_in(my_x, my_y)) {``), allowing the returned messages to be iterated.
+The spatial message iterator is accessed using :func:`FLAMEGPU->message_in()<flamegpu::MessageSpatial2D::In::operator()>` (or via the ``message_in`` agent function argument in Python), this takes two ``float`` parameters specifying the search origin. Normally this will be passed directly to a C++ range-based for loop, allowing the returned messages to be iterated.
 
-In the case of :class:`MessageSpatial2D<flamegpu::MessageSpatial2D>`, the returned :class:`Message<flamegpu::MessageSpatial2D::In::Filter::Message>` objects only provide :func:`getVariable()<template<typename T, unsigned int N> __device__ T flamegpu::MessageSpatial2D::In::Filter::Message::getVariable(const char(&)[N]) const>` methods for returning the variables and array variables stored within the message.
+In the case of :class:`MessageSpatial2D<flamegpu::MessageSpatial2D>`, the returned :class:`Message<flamegpu::MessageSpatial2D::In::Filter::Message>` objects only provide :func:`getVariable()<template<typename T, unsigned int N> __device__ T flamegpu::MessageSpatial2D::In::Filter::Message::getVariable(const char(&)[N]) const>` methods for returning the variables and array variables stored within the message. The *pure* Python equivalent requires appended Types and array lengths.
```

Comment:
>  The *pure* Python equivalent requires appended Types and array lengths

requires the type and array length to be appended to the name?

Reply:
fixed